analyzer/codechecker_analyzer/cmd/parse.py

```diff
@@ -755,8 +790,47 @@ def trim_path_prefixes_handler(source_file):
         """
         return util.trim_path_prefixes(source_file, trim_path_prefixes)
 
+    class ReportCounter:
+        """ Functor class to count number of reports of subsequent calls of
+        skip_html_report_data_handler() function."""
+
+        def __init__(self, initial=0) -> None:
+            """ Constructor.
+            Parameters
+            ----------
+                initial : Initial value of counter.
+            """
+
+            self.__sum = initial
+
+        def __call__(self, result):
+            """ Decorator function for bug counting.
+            Parameters
+            ----------
+                result : The result of call of the decorated function.
+            """
+            if not result[0]:  # Bug report was not skipped
+                self.__sum += 1
+
+        @property
+        def sum(self) -> None:
+            """ Counter getter. """
+            return self.__sum
+
+        @sum.setter
+        def sum(self, initial) -> None:
+            """ Counter setter.
+            Parameters
+            ----------
+                initial : Initial value of counter.
+            """
+
+            self.__sum = initial
+
     html_builder = None
+    report_counter = ReportCounter()
 
+    @ResultVisitor(report_counter)
```

Comment:
As far as I see we use the `ResultVisitor` and `ReportCounter` classes only here to get the number of non-skipped report counts and also the `ReportCounter` class expects the same return type as the `skip_html_report_data_handler` function.

I think you can use the following solution to do the same thing which is much shorter (2 lines vs ~50 lines source code):
```py
num_of_reports: int = 0
processed_path_hashes: Set[str] = set()
def skip_html_report_data_handler():
    ...
    if not skip:
        processed_path_hashes.add(path_hash)
        num_of_reports += 1
    ...
```

If you check this function we do the same with `processed_path_hashes`.

Reply:
Your suggestion introduces a hidden parameter of skip_html_report_data_handler() callback function. This behaves as "global variables". I think that increases entropy of the code. The more detailed overview of this topic can be found in this article: https://stackoverflow.com/questions/19158339/why-are-global-variables-evil